cpp/src/io/parquet/page_data.cu

```diff
@@ -1812,10 +1803,16 @@ void PreprocessColumnData(hostdevice_vector<PageInfo>& pages,
 
   // computes:
   // PageNestingInfo::size for each level of nesting, for each page.
-  // The output from this does not take row bounds (num_rows, min_row) into account
+  // This computes the size for the entire page, not taking row bounds into account.
+  // If row_bounds is set to true, we have to do a second pass later that "trims"
+  // the starting and ending read values to account for these bounds.
   gpuComputePageSizes<<<dim_grid, dim_block, 0, stream.value()>>>(
-    pages.device_ptr(), chunks.device_ptr(), min_row, num_rows, chunks.size(), false);
-  stream.synchronize();
+    pages.device_ptr(),
+    chunks,
+    // if row_bounds is false, include all possible rows.
+    row_bounds ? min_row : 0,
+    row_bounds ? num_rows : INT_MAX,
```

Comment:
Should min_row and num_rows be `std::optional` instead of this extra  `row_bounds` bool?

Reply:
Yeah I like that idea. A std::optional pair would encapsulate all 3 variables.

Reply:
So I changed my mind here.   Here's the issue:  `num_rows` gets set to be a real value regardless of whether the user has specified row bounds by the time we get here.  So it doesn't make sense to bury it under an optional.  `skip_rows` can be checked, but the user could have left it as 0 but capped `num_rows`.  

In the calling function, the non-preprocess path expects that total_rows is valid/correct, so again an optional doesn't make sense.
https://github.com/rapidsai/cudf/blob/d4809a10b6f587dfb5155e6fe3b72d7091c32473/cpp/src/io/parquet/reader_impl.cu#L1368

Reply:
Worth pointing out:  this is why we need the whole trim pass in the first place. All we know is that starting row and total # of rows to read.  So we first need to run through each and figure out how many rows each page has (which can be < 1, amusingly).  And only after that can we go through and figure out where the start/end points really are.

Reply:
I understand. Thanks for investigating into this option.